### ISSUE TITLE
Enclosure routing overwrite min level

### ISSUE THREAD

**Issue by osboele** [valskalla/odin#84]
**Given**

Example for class and enclosure routing with odin 0.6.0

```
package net.boele.geo

import cats.implicits._
import zio._
import io.odin.consoleLogger
import io.odin.formatter.Formatter
import io.odin.Level
import io.odin.Logger
import io.odin.config._
import zio.interop.catz._
import zio.interop.catz.implicits._


object OdinApp extends App {

  val logger: Logger[Task] = 
    consoleLogger(Formatter.colorful, Level.Info)

  val enclosureLogger: Logger[Task] =
    enclosureRouting(
      "net.boele.geo.OdinApp.foo" -> consoleLogger[Task]().withMinimalLevel(Level.Warn),
      "net.boele.geo.OdinApp.bar" -> consoleLogger[Task]().withMinimalLevel(Level.Info),
      "net.boele" -> consoleLogger[Task]()
    ).withNoopFallback

  def zoo: Task[Unit] = enclosureLogger.debug("Debug")
  def foo: Task[Unit] = enclosureLogger.info("Never shown")
  def bar: Task[Unit] = enclosureLogger.warn("Warning")

  val prog = 
    for {
      _ <- logger.info("start app")
      _ <- zoo *> foo *> bar
      _ <- logger.info("done")
    } yield ()
  

  def run(args: List[String]): ZIO[ZEnv,Nothing,Int] =
    prog.fold(_ => 1, _ => 0)
}
```

**When** executed

**Expect**

"Never shown" should not be shown

**Actual**

```
2020-01-31T19:42:18,308 [zio-default-async-1-1227229563] INFO net.boele.geo.OdinApp.prog:32 - start app
2020-01-31T19:42:18,343 [zio-default-async-1-1227229563] DEBUG net.boele.geo.OdinApp.zoo:26 - Debug
2020-01-31T19:42:18,344 [zio-default-async-1-1227229563] INFO net.boele.geo.OdinApp.foo:27 - Never shown
2020-01-31T19:42:18,345 [zio-default-async-1-1227229563] WARN net.boele.geo.OdinApp.bar:28 - Warning
2020-01-31T19:42:18,347 [zio-default-async-1-1227229563] INFO net.boele.geo.OdinApp.prog:34 - done
```

Cause

Log-Level isn't checked by [`logger.log()`](https://github.com/valskalla/odin/blob/4506272d112904e8c27874cdebbe95fa0e90e45f/core/src/main/scala/io/odin/config/EnclosureRouting.scala#L42)

**Comment by sergeykolbasov** [comment]
Hi @osboele 

Thanks for the report, I'll take care of it

**Comment by sergeykolbasov** [comment]
Fixed in #85 
The root cause was exactly the one described.

Just published 0.6.1 release, should be available in maven soon

**Comment by osboele** [comment]
Hi @sergeykolbasov 

Thanks a lot!
